cmd/gardenlet/app/migration.go

```diff
@@ -36,6 +44,46 @@ func (g *garden) runMigrations(ctx context.Context, log logr.Logger, _ cluster.C
 	return nil
 }
 
+// TODO(timuthy): Remove after v1.97 was released.
+// updateGardenerResourceManager updates all GRM deployments in a seed cluster,
+// in order to roll out the Brotli compression support added in Gardener v1.97.
+func updateGardenerResourceManager(ctx context.Context, log logr.Logger, cl client.Client) error {
+	image, err := imagevector.ImageVector().FindImage(imagevector.ImageNameGardenerResourceManager)
+	if err != nil {
+		return err
+	}
+	image.WithOptionalTag(version.Get().GitVersion)
+
+	grmDeployments := &appsv1.DeploymentList{}
+	if err := cl.List(ctx, grmDeployments, client.MatchingLabels(map[string]string{
+		v1beta1constants.LabelApp:   v1beta1constants.DeploymentNameGardenerResourceManager,
+		v1beta1constants.GardenRole: v1beta1constants.GardenRoleControlPlane,
+	})); err != nil {
+		return err
+	}
+
+	fns := make([]flow.TaskFn, 0, len(grmDeployments.Items))
+	for _, grmDeployment := range grmDeployments.Items {
+		for j, container := range grmDeployment.Spec.Template.Spec.Containers {
+			if container.Name == v1beta1constants.DeploymentNameGardenerResourceManager ||
+				grmDeployment.Spec.Template.Spec.Containers[j].Image == image.String() {
+				continue
+			}
+
+			fns = append(fns, func(ctx context.Context) error {
+				patch := client.StrategicMergeFrom(grmDeployment.DeepCopy())
+				grmDeployment.Spec.Template.Spec.Containers[j].Image = image.String()
+
+				log.Info("Updating Gardener-Resource-Manager", "deployment", client.ObjectKeyFromObject(&grmDeployment))
+				return cl.Patch(ctx, &grmDeployment, patch)
+			})
+			break
+		}
+	}
+
+	return flow.Parallel(fns...)(ctx)
+}
```

Comment:
Why is the `seed-is-garden` handling no longer present compared to the reverted change? Do we no longer need it?

Reply:
As discussed via chat, it should be sufficient to update the shoot GRMs because there it's primarily necessary for the ones in a FAILED state.